controlpanel/api/auth0.py

```diff
@@ -127,15 +127,63 @@ def get_group(self, group_name):
 
         return groups[0]
 
-    def get_group_members(self, group_name):
+    def get_group_id(self, group_name):
         group = self.get_group(group_name)
         if group:
+            return group["_id"]
```

Comment:
If we're programming defensively here (I don't know the code, but if the `"_id"` key is missing [is that likely?] you'll throw a `KeyError` here) you could rewrite the line as: `return group.get("_id")` which will return `None` if the key doesn't exist. Since you're checking the return value of this method when you call it in the code below, a `None` will allow the app to continue without doing anything meaningful without causing a 500-ish error.

Reply:
Yeah in this case the `_id` key would always be there hence it's fine to have it blow up with `KeyError` when it's not. The Auth0 API and "contract" would be broken if this is missing so I'd be happy to raise an exception.

I've done what you've suggested [in another part](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/799/files#diff-4abd63b561b5260e074659dad215fa1cR166) where their API is behaving in an inconsistent way:

> role.get("permissions", [])

For some obscure reasons they decided to omit the `permissions` key when a Role didn't have any associated permission and include it when it did. In that case I definitely wanted to be more defensive and default to an empty list when that key was missing as semantically means "empty list of associated permissions" and we don't want to raise an exception in that case.

Sorry for the long comment, I hope it helps clarify ☺️